pegaprox/api/settings.py

```
@@ -2627,6 +2595,170 @@ def restore_config():
+        return jsonify({'error': safe_error(e, 'Provider could not be removed')}), 500
+
+
+@bp.route('/api/config/vault/providers/<provider_type>/test', methods=['POST'])
+@require_auth(roles=[ROLE_ADMIN])
+def config_vault_test_provider(provider_type):
+    try:
+        test_vault_provider(provider_type)
+        return jsonify({'success': True, 'message': 'Connection successful'})
+    except ValueError as e:
+        return jsonify({'error': str(e)}), 400
+    except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)